### PR TITLE
Topic/arrow scaladoc

### PR DESCRIPTION
**Pull request by lukewyman**
Work for #788. Added Scaladoc comments for cats/arrow/Arrow.scala.

### CODE REVIEW

**Comment by striation** [comment]
Looks great to me. :+1: 

I think that Travis error is spurious, we should retry it.

File: core/src/main/scala/cats/arrow/Arrow.scala [hunk 1]
```diff
@@ -5,11 +5,58 @@ import cats.functor.Strong
...
+    * scala> val dimapArrow = Arrow[Function1].dimap(fab)(f)(g)
+    * scala> dimapArrow(3)
+    * res0: Double = 5.4
+    * }}}
+    */
   def dimap[A, B, C, D](fab: F[A, B])(f: C => A)(g: B => D): F[C, D] =
     compose(lift(g), andThen(lift(f), fab))
 
+  /**
+    * Create a new arrow that takes two inputs, but only modifies the second input
```
Comment (ceedubs) [inline core/src/main/scala/cats/arrow/Arrow.scala]: typo: only modifies the _first_ input

**Comment by ceedubs** [comment]
Thanks, @lukewyman! Just one minor typo, and otherwise this looks great.

**Comment by lukewyman** [comment]
pushed a change to fix the typo on Arrow.first

**Comment by ceedubs** [comment]
Thanks! :+1:

**Pull request merged**